### PR TITLE
Document using AnimatableBody when needed in StaticBody description

### LINKED ISSUE #75358: Area2D has inconsistent interactions with StaticBody2D when the Area2D is moving or still
### Godot version

4.0.1 stable

### System information

MacOS 13.0, Forward+, M1 Metal

### Issue description

Area2D has an inconsistent interaction with StaticBody2Ds. When Area2D holds still, it can't see the moving StaticBody2D.

For my project, I'm scaling StaticBody2D for interactions between CharacterBody2D and Area2D. I discovered an inconsistency when these scaled StaticBody2Ds entered into the Area2D's collision bodies. The Area2D didn't send the body_entered(body) signal unless the Area2D was in motion. A simple rotation on the Area2D will do the trick. Here's a short video of my test case:

https://user-images.githubusercontent.com/6932210/227780426-52fe7c25-435f-4c9b-8076-b335733b500a.mov

My expectation is that an Area2D would detect the StaticBody2D whether or not the Area2D is in motion.

tl;dr:  StaticBody2D plays a game of "Red Light, Green Light" with Area2D. 

### Steps to reproduce

Play the sample scene called "scene" in the attached project.

For my simple example, I created an Area2D scene that turns red when a body enters it, and reverts back to white when the body exits. I scale a StaticBody2D using an AnimationPlayer node. Instancing the Area2D scene three times, one instance stays still, one rotates constantly, and another moves into the StaticBody2D then holds still and lets the StaticBody2d move out of its collision body on its own.

The Area2D that doesn't move ignores the StaticBody2D and never sends the appropriate signals.

The Area2D that moves into the StaticBody2D then holds still sends the appropriate signals. I was wondering if moving into the StaticBody2D would send the body_entered(body) signal, and if holding still wouldn't send the body_exited(body) signal, but that works as expected.

The rotating Area2D also successfully sends the body_entered(body) and body_exited(body).

### Minimal reproduction project

[Areas and Static Bodies.zip](https://github.com/godotengine/godot/files/11071737/Areas.and.Static.Bodies.zip)

### PR DESCRIPTION
**Pull request by Calinou**
- This closes https://github.com/godotengine/godot/issues/75358.

### CODE REVIEW

File: doc/classes/StaticBody2D.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
 		Static body for 2D physics.
 		A static body is a simple body that doesn't move under physics simulation, i.e. it can't be moved by external forces or contacts but its transformation can still be updated manually by the user. It is ideal for implementing objects in the environment, such as walls or platforms. In contrast to [RigidBody2D], it doesn't consume any CPU resources as long as they don't move.
 		They have extra functionalities to move and affect other bodies:
-		[b]Static transform change:[/b] Static bodies can be moved by animation or script. In this case, they are just teleported and don't affect other bodies on their path.
+		[b]Static transform change:[/b] Static bodies [i]can[/i] be moved by animation or script. In this case, they are just teleported and don't affect other bodies on their path. Use [AnimatableBody2D] instead of [StaticBody2D] if you need a moving static body that affects other bodies on its path.
```
Comment (bprzybytkowski) [inline doc/classes/StaticBody2D.xml:10]: WDYT about somehow mentioning the "not affecting other bodies" part inside `<brief_description>` as well? IMO current phrasing implies that StaticBodies can be used to create programatically moving walls or platforms while in reality they shouldn't. The same goes for StaticBody3D

File: doc/classes/StaticBody3D.xml [hunk 2]
```diff
@@ -1,14 +1,14 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="StaticBody3D" inherits="PhysicsBody3D" version="4.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
 	<brief_description>
-		Physics body for 3D physics which is static or moves only by script. Useful for floor and walls.
+		Physics body for 3D physics which is static or moves only by script (without affecting other bodies on its path). Useful for floor and walls.
```
Comment (YuriSizov) [inline doc/classes/StaticBody3D.xml:4]:
```suggestion
		Physics body for 3D physics which is static or moves only by script (without affecting other bodies on its path). Useful for floors and walls.
```

For consistency :)

**Review by YuriSizov — APPROVED** [review]

**Comment by YuriSizov** [comment]
@Calinou Something went wrong with your rebase

**Comment by YuriSizov** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.3.
